pkg/reconciler/trigger/trigger.go

```diff
@@ -301,3 +365,25 @@ func (r *Reconciler) reconcileSubscription(ctx context.Context, t *v1alpha1.Trig
 	}
 	return newSub, nil
 }
+
+func getObjRefFromDependencyAnnotation(dependencyAnnotation string) (corev1.ObjectReference, error) {
+	var objectRef corev1.ObjectReference
+	if err := json.Unmarshal([]byte(dependencyAnnotation), &objectRef); err != nil {
+		return objectRef, err
+	}
+	return objectRef, nil
+}
+
+func validateDependencyAnnotation(depObjRef corev1.ObjectReference) error {
+	var result *multierror.Error
+	if depObjRef.Namespace == "" {
```

Comment:
I would say that the namespace _has_ to be empty. Then we always force it to be the same namespace as the Trigger. 

Allowing cross-namespace dependency tracking opens huge RBAC escalation problems (e.g. I, without any permissions on the prod namespace, can set a dependency to something in that namespace and see its readiness).

Reply:
Another issue with cross-namespace dependency tracking: GC doesn't work across namespaces